Makefile

```diff
@@ -14,6 +14,24 @@ KUBECONFIG ?= $(HOME)/.kube/config
 GO_PACKAGES = ./cmd/... ./pkg/...
 GO_FILES := $(shell find $(shell $(GO) list -f '{{.Dir}}' $(GO_PACKAGES)) -name \*.go)
 
+COMMIT = $(shell git rev-parse HEAD)
+TAG = $(shell git describe --exact-match --abbrev=0 --tags '$(COMMIT)' 2> /dev/null || true)
+DIRTY = $(shell git diff --shortstat 2> /dev/null | tail -n1)
+
+# Use a tag if set, otherwise use the commit hash
+ifeq ($(TAG),)
+VERSION := $(COMMIT)
+else
+VERSION := $(TAG)
+endif
+
+# Check for changed files
+ifneq ($(DIRTY),)
+VERSION := $(VERSION)+dirty
+endif
+
+GO_LD_FLAGS = -ldflags "-X main.VERSION=$(VERSION)"
```

Comment:
From experience, this variable is impossible to extend, because `go` chokes on multiple `-ldflags`.  I suggest instead moving the actual `-ldflags` flag up one level:

```Makefile
GO_LD_FLAGS = -X main.VERSION=$(VERSION)

kubeseal: $(GO_FILES)
	$(GO) build -o $@ $(GO_FLAGS) -ldflags "$(GO_LD_FLAGS)" ./cmd/kubeseal
```